### PR TITLE
[CGP-389] [Docs] Added a writeup on mutual type-level recursion

### PR DESCRIPTION
**Pull request by effectfully**

### CODE REVIEW

File: docs/fomega/mutual-type-level-recursion/MutualData.agda [hunk 1]
```diff
@@ -0,0 +1,420 @@
...
+
+  Forestᵗ : TreeForestᵗ
+  Forestᵗ A B = B
+
+  -- Where `TreeForestᵗ` is the type of inlined matchers which always return a `Set` and
+  -- `Treeᵗ` and `Forestᵗ` are such matchers. The rest is straightforward:
+
+  TreeForest : Set -> TreeForestᵗ -> Set
+  TreeForest A =
+    IFix λ R tfᵗ ->
```
Comment (michaelpj) [inline docs/fomega/mutual-type-level-recursion/MutualData.agda]: Can we call this `Rec` just for consistency?

Comment (effectfully) [reply docs/fomega/mutual-type-level-recursion/MutualData.agda]: Err, wanted to replace `R` by `Rec` everywhere and forgot. Thanks.

**Comment by effectfully** [comment]
I'd prefer for Philip to take a loop first. Should have added the "DO NOT MERGE" label. Well, not too important as these are just docs that can be extended/corrected in a future PR if necessary.

**Comment by michaelpj** [comment]
I think it's fine to have it in, since I think it's correct, it's just a question of whether or not we will use it.

**Pull request merged**